examples/OpenTracing.Examples/LateSpanFinish/LateSpanFinishTest.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OpenTracing.Mock;
+using Xunit;
+
+using static OpenTracing.Examples.TestUtils;
+
+namespace OpenTracing.Examples.LateSpanFinish
+{
+    public class LateSpanFinishTest
+    {
+        private readonly MockTracer _tracer = new MockTracer();
+
+        [Fact]
+        public void test()
+        {
+            // Create a Span manually and use it as parent of a pair of subtasks
+            ISpan parentSpan = _tracer.BuildSpan("parent").Start();
+            using (IScope scope = _tracer.ScopeManager.Activate(parentSpan, finishSpanOnDispose:false))
+            {
+                SubmitTasks();
+            }
+
+            WaitForSpanCount(_tracer, 2, DefaultTimeout);
+
+            // Late-finish the parent Span now
+            parentSpan.Finish();
+
+            var spans = _tracer.FinishedSpans();
+            Assert.Equal(3, spans.Count);
+            Assert.Equal("task1", spans[0].OperationName);
+            Assert.Equal("task2", spans[1].OperationName);
+            Assert.Equal("parent", spans[2].OperationName);
+
+            TestUtils.AssertSameTrace(spans);
+
+            Assert.Null(_tracer.ActiveSpan);
+        }
+
+        // Fire away a few subtasks, passing a parent ISpan whose lifetime
+        // is not tied at-all to the children.
+        // NOTE: As opposed to Java, there is not need to reactivate the parent Span,
+        // as the context is propagated by AsyncLocalScopeManager.
+        private void SubmitTasks()
```

Comment:
I think you could make this method return a Task and store both tasks and just return "Task.WhenAll(task1, task2)". You can then just `await` the call and don't have to call "WaitForSpanCount".